mav_trajectory_generation/src/polynomial.cpp

```diff
@@ -196,6 +196,12 @@ bool Polynomial::getPolynomialWithAppendedCoefficients(
   }
 }
 
+void Polynomial::scalePolynomialInTime(double scaling_factor) {
+  for (int n = 0; n < N_; n++) {
+    coefficients_[n] *= pow(scaling_factor, n);
```

Comment:
pow is doing a lot of double work, but then again the polynomials are small so I guess this is super fast